src/materialized/src/bin/materialized/main.rs

```diff
@@ -423,6 +442,76 @@ fn main() {
     }
 }
 
+fn finish_and_init_tracing<
+    L: tracing_subscriber::layer::Layer<S> + Send + Sync + 'static,
+    S: ::tracing::Subscriber + Send + Sync + 'static,
+>(
+    stack: tracing_subscriber::layer::Layered<L, S>,
+    runtime: &tokio::runtime::Runtime,
+    opentelemetry_endpoint: Option<String>,
+    opentelemetry_headers: Option<String>,
+) -> Result<(), anyhow::Error>
+where
+    tracing_subscriber::layer::Layered<L, S>: tracing_subscriber::util::SubscriberInitExt,
+    for<'ls> S: tracing_subscriber::registry::LookupSpan<'ls>,
+{
+    use tracing_subscriber::util::SubscriberInitExt;
+
+    if let Some(endpoint) = opentelemetry_endpoint {
+        use tracing_subscriber::layer::SubscriberExt;
+        // Setting up opentel in the background requires we are in a tokio-runtime
+        // context
+        let _guard = runtime.enter();
+
+        // Manually setup an openssl-backed, h2, proxied `Channel`
+        // let endpoint = tonic::transport::Endpoint::from_static("https://api.honeycomb.io/");
+        let endpoint = tonic::transport::Endpoint::from_shared(endpoint)?;
+        let channel = endpoint
+            .connect_with_connector_lazy(mz_http_proxy::hyper::connector_with_h2_alpn())
+            .unwrap();
+        let mut mmap = tonic::metadata::MetadataMap::new();
+        /*
+        mmap.append(
+            "x-honeycomb-team",
+            "f00cbe54a4b49537c987dc6a0e833564".parse().unwrap(),
```

Comment:
Just to make sure this is recognized as a fake key:

```suggestion
            "fffffffffffffffffffffffffffffffff".parse().unwrap(),
```

Reply:
ugh, so embarrassing